library/artifactory_repo.py

```diff
@@ -103,15 +107,60 @@
     state:
         description:
             - The state the repository should be in. 'present' ensures that a
-              repository is created and/or it is updated, but not replaced.
-              'absent' ensures that the repository is deleted. 'read' will
-              return the configuration if the repository exists.
+              repository exists, but not replaced.
+              'update' will attempt to update the configuration of the remote
```

Comment:
Why did we introduce the update state?

Reply:
It allows us to do validation in check_mode, otherwise the module will have to perform an api call to determine if the repo exists:
if exists -> update
if not exists -> create

Reply:
There is a problem with the code, if that's what you're talking about, but that is an easy fix.